examples/others/plot_learning_weights_with_COOT.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+"""
+=================================
+Learning sample marginal distribution with CO-Optimal Transport
+=================================
+
+In this example, we illustrate how to estimate the sample marginal distribution which minimizes the CO-Optimal Transport distance [47] between two matrices. More precisely, given a source data $(X, \mu_x^{(s)}, \mu_x^{(f)})$ and a target matrix $Y$ associated with a fixed histogram on features $\mu_y^{(f)}$, we want to solve the following problem
+$$
+\min_{\mu_y^{(s)} \in \Delta} \text{COOT}\left( (X, \mu_x^{(s)}, \mu_x^{(f)}), (Y, \mu_y^{(s)}, \mu_y^{(f)}) \right)
+$$
+where $\Delta$ is the probability simplex. This minimization is done with a simple projected gradient descent in PyTorch. We use the automatic backend of POT that allows us to compute the CO-Optimal Transport distance with `ot.coot.co_optimal_transport2` with differentiable losses.
+
+[47] Redko, I., Vayer, T., Flamary, R., and Courty, N. (2020). [CO-Optimal Transport](https://proceedings.neurips.cc/paper/2020/file/cc384c68ad503482fb24e6d1e3b512ae-Paper.pdf). Advances in Neural Information Processing Systems, 33.
+"""
+
+# Author: Remi Flamary <remi.flamary@unice.fr>
+#         Quang Huy Tran <quang-huy.tran@univ-ubs.fr>
+# License: MIT License
+
+from matplotlib.patches import ConnectionPatch
+import torch
+import numpy as np
+
+import matplotlib.pyplot as pl
+import ot
+
+from ot.coot import co_optimal_transport as coot
+from ot.coot import co_optimal_transport2 as coot2
+
+"""# Generate data
+
+The source and clean target matrices are generated by
+$X_{i,j} = \cos(\frac{i}{n_1} \pi) + \cos(\frac{j}{d_1} \pi)$ and $Y_{i,j} = \cos(\frac{i}{n_2} \pi) + \cos(\frac{j}{d_2} \pi)$. The target matrix is then contaminated by adding $5$ row outliers. Intuitively, we expect that the estimated sample distribution should ignore these outliers, i.e. their weights should be zero.
+"""
+
+np.random.seed(182)
+
+n1, d1 = 20, 16
+n2, d2 = 10, 8
+n = 15
+
+X = torch.cos(torch.arange(n1) * torch.pi /
+              n1)[:, None] + torch.cos(torch.arange(d1) * torch.pi / d1)[None, :]
+# generate clean target data mixed with outliers
+Y_noisy = torch.randn((n, d2)) * 10.0
+Y_noisy[:n2, :] = torch.cos(torch.arange(
+    n2) * torch.pi / n2)[:, None] + torch.cos(torch.arange(d2) * torch.pi / d2)[None, :]
+Y = Y_noisy[:n2, :]
+
+X, Y_noisy, Y = X.double(), Y_noisy.double(), Y.double()
+
+fig, ax = pl.subplots(nrows=1, ncols=3, figsize=(12, 5))
+ax[0].imshow(X, vmin=-2, vmax=2)
+ax[0].set_title('$X$')
+
+ax[1].imshow(Y, vmin=-2, vmax=2)
+ax[1].set_title('Clean $Y$')
+
+ax[2].imshow(Y_noisy, vmin=-2, vmax=2)
+ax[2].set_title('Noisy $Y$')
+
+pl.tight_layout()
+
+"""## Optimize the COOT distance with respect to the sample marginal distribution"""
```

Comment:
please see how section text blocks should be formatted for sphinx-gallery to work